src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```
@@ -0,0 +1,111 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private final RegistrationServiceImpl service = new RegistrationServiceImpl();
+
+    @Test
+    void register_null_notOK() {
+        User actual = null;
+        assertThrows(IllegalArgumentException.class, () -> service.register(actual));
```

This test expects IllegalArgumentException for a null user. The task and checklist require using a custom unchecked exception for validation failures and explicit guidance to avoid NullPointerException/IllegalArgumentException for validation. Quote from checklist: "Don't use `NullPointerException`, because in such case we can't distinguish was this exception because of our mistake or because of validation." and the description requires: "You should create your custom unchecked exception and throw it in case of invalid data." Update the test to expect RegistrationException (or a specific subclass) and ensure the implementation throws the custom exception instead of IllegalArgumentException/NullPointerException.